### PR TITLE
Replace the StatsWriter plugin with better performing BundleAnalyzerPlugin

### PR DESCRIPTION
**Pull request by jsnajdr**
For generating the `stats.json` file, remove our own `StatsWriter` that uses `JSON.stringify` and replace it with `BundleAnalyzerPlugin` that uses the `bfj` library to write the JSON file as a stream, consuming less memory, and is built into the `webpack-bundle-analyzer` NPM package.

Also, webpack doesn't need the `profile` flag when emitting stats: we don't use the timing information there when analyzing the bundles.

This change should reduce webpack memory requirements when generating the stats.

#### Testing instructions

Verify that `npm run analyze-bundes` still works as expected. ICFY on this branch reports "No changes in production JS", which is a good sign :wink:

Can we somehow verify what is the memory consumption of the CircleCI `icfy-stats` job before and after this PR?

### CODE REVIEW

**Review by sirreal — DISMISSED** [review]
I'm approving with Canary failures. I expect those are unrelated, but please make sure before merging.

The ICFY job on CircleCI is producing artifacts and I see they have arrived on ICFY correctly, which reports no changes to production JS.

Nice to remove that custom plugin 👍

**Comment by jsnajdr** [comment]
> I'm approving with Canary failures. I expect those are unrelated, but please make sure before merging.

They are related :sad: Tests fail because the calypso.live branch is not live because Calypso fails to build because `webpack-bundle-analyzer` is now required by webpack config and is a dev dependency -- not installed by the production Docker build.

Pushed a commit that (hopefully) fixes that.

**Review by sirreal — APPROVED** [review]
calypso.live is now working well and canaries are passing.

CircleCI still appears to produce artifacts correctly:

```
Uploading /tmp/artifacts to tmp/artifacts
Uploading /tmp/artifacts/icfy/chart.json (7.8 MB): DONE
Uploading /tmp/artifacts/icfy/stats.json (363 MB): DONE
```

**Pull request merged**
